certbot/le_util.py

```diff
@@ -209,9 +209,75 @@ def safely_remove(path):
             raise
 
 
-def get_os_info():
+def get_os_info(filepath="/etc/os-release"):
+    """
+    Get OS name and version
+
+    :param str filepath: File path of os-release file
+    :returns: (os_name, os_version)
+    :rtype: `tuple` of `str`
+    """
+
+    if os.path.isfile(filepath):
+        # Systemd os-release parsing might be viable
+        os_name, os_version = get_systemd_os_info(filepath=filepath)
+        if os_name:
+            return (os_name, os_version)
+
+    # Fallback to platform module
+    return get_python_os_info()
+
+
+def get_systemd_os_info(filepath="/etc/os-release"):
+    """
+    Parse systemd /etc/os-release for distribution information
+
+    :param str filepath: File path of os-release file
+    :returns: (os_name, os_version)
+    :rtype: `tuple` of `str`
+    """
+
+    os_name = _get_systemd_os_release_var("ID", filepath=filepath)
+    os_version = _get_systemd_os_release_var("VERSION_ID", filepath=filepath)
+
+    return (os_name, os_version)
+
+
+def _get_systemd_os_release_var(varname, filepath="/etc/os-release"):
+    """
+    Get single value from systemd /etc/os-release
+
+    :param str varname: Name of variable to fetch
+    :param str filepath: File path of os-release file
+    :returns: requested value
+    :rtype: `str`
+    """
+
+    var_string = varname+"="
+    if not os.path.isfile(filepath):
+        return ""
+    with open(filepath, 'r') as fh:
+        contents = fh.readlines()
+
+    for line in contents:
```

Comment:
I do like it better too, however .startswith() makes it more readable in my opinion, if someone has to look into / touch that in the future. Also the comment is in place to account for the unreadable part. Being able to quickly spot why we're getting in to that if clause makes up for it imo.